stdlib/public/core/MemoryLayout.swift

```diff
@@ -153,10 +183,36 @@ extension MemoryLayout {
   /// - Returns: The default memory alignment, in bytes, of the given value's
   ///   type. This value is always positive.
   @_transparent
-  public static func alignment(ofValue value: T) -> Int {
+  @_alwaysEmitIntoClient
+  public static func alignment(ofValue value: borrowing T) -> Int {
     return MemoryLayout.alignment
   }
+}
 
+extension MemoryLayout {
+  // TODO: Merge this back into the noncopyable variant once we have @_preInverseGenerics
+  @available(swift, obsoleted: 5.11) // Legacy ABI compatibility
+  @usableFromInline
+  internal static func size(ofValue value: borrowing T) -> Int {
+    return MemoryLayout.size
+  }
+
+  // TODO: Merge this back into the noncopyable variant once we have @_preInverseGenerics
+  @available(swift, obsoleted: 5.11) // Legacy ABI compatibility
+  @usableFromInline
+  internal static func stride(ofValue value: borrowing T) -> Int {
+    return MemoryLayout.stride
+  }
+
+  // TODO: Merge this back into the noncopyable variant once we have @_preInverseGenerics
+  @available(swift, obsoleted: 5.11) // Legacy ABI compatibility
+  @usableFromInline
+  internal static func alignment(ofValue value: borrowing T) -> Int {
+    return MemoryLayout.alignment
+  }
+}
+
+extension MemoryLayout {
```

Comment:
It may be good practice to start writing these things `extension MemoryLayout where T: Copyable` for explicitness. Subjective though, and is my opinion x)